pdns/dnsdist-lua-inspection.cc

```diff
@@ -31,26 +32,27 @@ static std::unordered_map<unsigned int, vector<boost::variant<string,double>>> g
   map<DNSName, unsigned int> counts;
   unsigned int total=0;
   {
-    std::lock_guard<std::mutex> lock(g_rings.respMutex);
-    if(!labels) {
-      for(const auto& a : g_rings.respRing) {
-        if(!pred(a))
-          continue;
-        counts[a.name]++;
-        total++;
-      }
-    }
-    else {
-      unsigned int lab = *labels;
-      for(auto a : g_rings.respRing) {
-        if(!pred(a))
-          continue;
-
-        a.name.trimToLabels(lab);
-        counts[a.name]++;
-        total++;
+    for (auto& shard : g_rings.d_shards) {
+      std::lock_guard<std::mutex> rl(shard->respLock);
+      if(!labels) {
+        for(const auto& a : shard->respRing) {
+          if(!pred(a))
+            continue;
+          counts[a.name]++;
+          total++;
+        }
       }
+      else {
+        unsigned int lab = *labels;
+        for(auto a : shard->respRing) {
```

Comment:
maybe `auto& a` ?

Reply:
We can't because we modify the `name` member by calling `DNSName::trimToLabels()` on it below. We could consider only copying the `DNSName` instead of the whole entry, but I'm not sure it would make a noticeable difference. It might worth benchmarking it, or simply doing it.